app/__init__.py

```diff
@@ -19,11 +22,44 @@ def create_app(config=None):
         # to override the app's default settings for testing
         app.config.update(config)
 
-    db.init_app(app)
-    migrate.init_app(app, db)
+    try:
+        db.init_app(app)
+        migrate.init_app(app, db)
+        logger.info("Database initialized successfully.")
+
+        with app.app_context():
+            db.session.execute("SELECT 1")
+            logger.info("DB connection test passed")
+    except Exception as e:
+        logger.exception("Error during DB initialization or connection")
```

Comment:
🔎 Notice the use of `logger` up here will result in crashing the startup. `logger` isn't available until line 43.